### ISSUE TITLE
Dragon Age II [BLUS30645] - Frequently hangs

### ISSUE THREAD

**Issue by Shoegzer** [RPCS3/rpcs3#16087]
### Quick summary

This game hangs frequently, observed even during the opening sequence.

### Details

The following settings were changed from default to attempt resolution, to no avail:

* SPU Decoder: Recompiler (ASMJIT)
* Driver Wave-up Delay: 200 | 800
* RSX FIFO Accuracy: Atomic

Please feel free to suggest any other settings that might help.

### Attach a log file

[RPCS3.log](https://github.com/user-attachments/files/17022121/RPCS3.log)

### System configuration

* OS: Linux Mint 22 / kernel 6.8
* CPU: Intel i7-10870H
* GPU: Nvidia GTX-3070 / driver 535.183
* Commit tested: 7f30e52f

### Other details

This may be the same issue affecting #10541

**Comment by Shoegzer** [comment]
Update: Setting Driver Wake-Up Delay to 2400 in the custom config file _config_BLUS30645.yml_ seems so far to work around the hanging issue with Vulkan (the UI only allows up to 800 which is not high enough).  This is the same workaround for Dragon Age: Origins as noted in the issue linked above.  

OpenGL has been reported to work too, though this option appears to be broken on Linux systems (at least on this test system rpcs3 has always frozen when compiling shaders on any game no matter what Nvidia driver is used).

**Comment by Triticum0** [comment]
had the same issue last year when testing the demo. don't know if the log will help.
[link](https://forums.rpcs3.net/showthread.php?tid=206446&highlight=Dragon+Age+II+Demo)

**Comment by lazyhax** [comment]
Can confirm, I have the same hanging issues on my ROG Ally. I can provide the logs if additional info is required.

**Comment by ValentinnoCruz** [comment]
Having the same issue on my Win 10

**Comment by digant73** [comment]
@Shoegzer, @ValentinnoCruz is that still an issue?

**Comment by Shoegzer** [comment]
Yes it is, just tested on latest commit.  It hangs very quickly ingame, even with driver wake-up delay set at 2400 in the custom config.

**Comment by Arka161** [comment]
The issue still occurs.

**Comment by Enriador** [comment]
Can confirm the game still inevitably freezes as of the latest version (v0.0.38-18588-d929596b Alpha).

**Comment by Darkhost1999** [comment]
> OpenGL has been reported to work too

[RPCS3.log](https://github.com/user-attachments/files/24545399/RPCS3.log)
Hang when using OpenGL. I was naming my character.

**Comment by Darkhost1999** [comment]
[RPCS3.log.gz](https://github.com/user-attachments/files/25011907/RPCS3.log.gz)
Debug output enabled

E {RSX [0x0061e0c]} RSX: ERROR: [Validation] Code 0 : Validation Error: [ VUID-VkImageMemoryBarrier-oldLayout-01197 ] Object 0: handle = 0x1bdb87c27f0, type = VK_OBJECT_TYPE_COMMAND_BUFFER; Object 1: handle = 0xf9bc0000077b22, type = VK_OBJECT_TYPE_IMAGE; | MessageID = 0x124ffb34 | vkCmdPipelineBarrier(): pImageMemoryBarriers[0].image (VkImage 0xf9bc0000077b22[]) cannot transition the layout of aspect=1, level=0, layer=0 from VK_IMAGE_LAYOUT_PRESENT_SRC_KHR when the previous known layout is VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL. The Vulkan spec states: If srcQueueFamilyIndex and dstQueueFamilyIndex define a queue family ownership transfer or oldLayout and newLayout define an image layout transition, oldLayout must be VK_IMAGE_LAYOUT_UNDEFINED or the current layout of the image subresources affected by the barrier (https://vulkan.lunarg.com/doc/view/1.3.268.0/windows/1.3-extensions/vkspec.html#VUID-VkImageMemoryBarrier-oldLayout-01197)

E SYS: The game did not react to the exit request in time. Terminating manually... 
MFC Events read aborted
SPU: Aborting unsaveable state

**Comment by Enriador** [comment]
Can confirm issue https://github.com/RPCS3/rpcs3/issues/16087 still happens as of 0.0.40-19164 (April 5, 2026).

I am having better results with OpenGL but things are still iffy. Will publish my settings soon.

**Comment by Enriador** [comment]
Okay, the game is more or less playable on OpenGL as of 0.0.40-19261 (April 28, 2026).

Pro tip: _always_ use OpenGL. _Never_ skip a cutscene line (for whatever reason it seems to drastically increase odds of game freezing in the next few moments during a cutscene). Finally, even with all this, the game might still freeze randomly in cutscenes, or even during the starting splash screen (a reboot or two normally fixes it for a while).

So it is _possible_ to play Dragon Age II on PC with native controller support. It's just way more painful than it needs to be.

My full settings for DA2: https://imgur.com/a/p1Fj2OT

**Comment by Shoegzer** [comment]
@Enriador What are your specs please (OS, CPU, GPU, GPU driver etc)? It would be better if you could list all your non-default settings inline with your post. Also is your driver wake-up delay set beyond 800 in the config?  The UI only allows setting to 800 for some reason, and this title is slightly more stable at much higher wake-up settings than that.

In testing against as many of your non-default config settings as I could find, this title consistently freezes whether you skip cutscenes or not, I think it's possible you're just getting lucky.

**Comment by Enriador** [comment]
@Shoegzer Windows 11, Ryzen 7 6800H, 680M (AMD). 

If I ever find the time and patience I will add the settings here instead of showing the images. To be quite honest, after 20+ hours of *just* testing DA2 on RPCS3 I lack both for the moment. lol

As said on the Imgur gallery:

"Might help to set Driver Wake-Up Delay to 2400 manually in the corresponding DA2 RPCS3 config file, config_NPUB30476.yml. I also used the custom Game Patch "Disable SPU MLAA", which is included with RPCS3."

>  it's possible you're just getting lucky

Not "possible", I effectively _am_ getting lucky. I can't emphasize it enough:

> even with all this, the game might still freeze randomly in cutscenes, or even during the starting splash screen (a reboot or two normally fixes it for a while).

Saving often and having a ton of patience can get you to the endgame, no doubt... as annoying as it can be. Hopefully RPCS3 devs will figure it out, one day!

**Comment by Shoegzer** [comment]
> If I ever find the time and patience I will add the settings here instead of showing the images. To be quite honest, after 20+ hours of just testing DA2 on RPCS3 I lack both for the moment. lol

Wow that's a lot.  I've spent several hours on it myself.

> As said on the Imgur gallery:

Missed that, sorry. The UI used to allow larger values, I don't know why it was force-limited to 800 as there are quite a few titles that require higher values, and you've probably noticed that any other changes to the UI reset the value to 800 in the config - inconvenient for testing and troubleshooting to say the least.

> Not "possible", I effectively am getting lucky. I can't emphasize it enough:

I just meant that whether you skip the cutscenes are not, I've observed that it makes no difference.  I think the conclusion that somehow it helps should be checked with others, because I think you're just getting lucky and if so, the conclusion is misleading.

> Saving often and having a ton of patience can get you to the endgame, no doubt... as annoying as it can be. Hopefully RPCS3 devs will figure it out, one day!

Hah well I don't have that kind of patience, but I'm glad you do and can report any issues ahead.  That said I _am_ patient for greater stability in RPCS3, I'm confident it'll be figured out eventually as you said.

**Comment by audiowylde2345** [comment]
How is this game playing being that in just a few months there's been tremendous updates and breakthroughs with this emulator?

**Comment by Enriador** [comment]
> I just meant that whether you skip the cutscenes are not, I've observed that it makes no difference

Dude I swear I am not bullshitting you, but just now I did 2 tries from the same save file (right after the ogre fight in the prologue, right before the cutscene where Leandra mourns Carver's death).

First try, I skipped 90% of the lines. Game froze in the Varric/Cassandra conversation.

Second try, didn't skip *anything*. Game went normally and I finally reached Kirkwall.

Ridiculously small sample size, I know. But when I get the time I might test it more times.

> How is this game playing being that in just a few months there's been tremendous updates and breakthroughs with this emulator?

Can you believe that - in spite of those massive changes RPCS3 has had recently, even in these 50 days since you commented with 75% of games now marked 'Playable' - that ABSOLUTELY NOTHING changed in regards to DA2?

Xenia, a far less developed emulator, finally fixed the hair flickering bug and DA2 + all its DLCs is now playable issue-free. https://youtube.com/watch?v=YBYJzAYajKA&pp=

**Comment by audiowylde2345** [comment]
I just tried it again always crashes after prologue after cut scene

**Comment by Enriador** [comment]
Use my settings: https://imgur.com/a/p1Fj2OT

Might take a few tries, but you WILL get past the prologue cutscene. Driver wake-up delay set to a high number is key.

**Comment by audiowylde2345** [comment]
I'll give it another shot

**Comment by audiowylde2345** [comment]
Maybe easier to stick with Xenia for now.

**Comment by audiowylde2345** [comment]
Yeah I tried what you said and it worked once but then it doesn't always work like you said. It's not worth the hassle at least to me. Until it's fixed I can easily play it on Xenia which just minor graphic glitches. Even though that emulator is not perfect it's good to have an option for now One day when it works I'll switch back it's no big deal Good thing is we're in a good time to have options so it's all gravy This emulator is on fire most of my games work there's only five or six that I've come across that don't work this being one of them

**Comment by Shoegzer** [comment]
Good news. After testing against the most recent master and playing through several scenes, and even using default settings, the freezing issue seems to be fully resolved.

There is a rendering issue where certain textures appear to "tear" particularly in the upper-left side of the screen, but that's not relevant to this writeup. 

So, another RPCS3 title can be marked as `Playable`!  I'll close this now, but can re-open if anyone finds a reason to.  Thanks to all those involved in testing.

**Comment by Darkhost1999** [comment]
Was able to play for a couple hours without crashing. So no problem here. Ryzen 7 7700x with integrated graphics, windows 11.

**Comment by audiowylde2345** [comment]
I tried it on my ally x I didn't play very much. Just wanted to see if it crashed before and after the character creation I can confirm I got further than before without crashing I'll play more over the next few days
